lib/index.js

```diff
@@ -354,10 +354,11 @@ class Mcdev {
      *
      * @param {string} businessUnit references credentials from properties.json
      * @param {string} type supported metadata type
-     * @param {string} customerKey Identifier of data extension
+     * @param {string} customerKey Identifier of metadata
+     * @param {number} [version] optional version of metadata
      * @returns {Promise.<void>} -
      */
-    static async deleteByKey(businessUnit, type, customerKey) {
+    static async deleteByKey(businessUnit, type, customerKey, version) {
```

Comment:
dont love that suddenly a journey concept becomes the default. would it not be better to use something like customerkey/version?

Reply:
it's an optional field after all - the type-classes have to implement using the version (or not)
I see your point, arent there any other types that might be able to use versions? :D

Reply:
as discussed - key/version construct

Reply:
done